cudaaligner/src/aligner_global_myers_banded.cpp

```diff
@@ -136,26 +121,62 @@ struct AlignerGlobalMyersBanded::InternalData
     batched_device_matrices<WordType> query_patterns;
 };
 
+void AlignerGlobalMyersBanded::reallocate_internal_data(InternalData* data, const int64_t max_device_memory, const int32_t max_bandwidth, const int32_t n_alignments_initial, cudaStream_t stream)
+{
+    const memory_distribution mem    = split_available_memory(max_device_memory, max_bandwidth);
+    DefaultDeviceAllocator allocator = data->seq_d.get_allocator();
+
+    data->seq_h.clear();
+    data->seq_starts_h.clear();
+    data->scheduling_index_h.clear();
+    data->results_h.clear();
+    data->result_lengths_h.clear();
+    data->result_starts_h.clear();
+
+    data->results_h.resize(mem.results_memory / sizeof(char));
+    data->seq_h.resize(2 * mem.sequence_memory / sizeof(char));
+
+    data->seq_starts_h.reserve(2 * n_alignments_initial + 1);
+    data->scheduling_index_h.reserve(n_alignments_initial);
+    data->result_lengths_h.reserve(n_alignments_initial);
+    data->result_starts_h.reserve(n_alignments_initial + 1);
+
+    data->seq_d.free();
+    data->seq_starts_d.free();
+    data->scheduling_index_d.free();
+    data->scheduling_atomic_d.free();
+    data->results_d.free();
+    data->result_starts_d.free();
+    data->result_lengths_d.free();
+    data->pvs            = batched_device_matrices<WordType>(0, allocator, stream);
+    data->mvs            = batched_device_matrices<WordType>(0, allocator, stream);
+    data->scores         = batched_device_matrices<int32_t>(0, allocator, stream);
+    data->query_patterns = batched_device_matrices<WordType>(0, allocator, stream);
+
+    data->seq_d.clear_and_resize(2 * mem.sequence_memory / sizeof(char));
+    data->seq_starts_d.clear_and_resize(2 * n_alignments_initial + 1);
+    data->scheduling_index_d.clear_and_resize(n_alignments_initial);
+    data->scheduling_atomic_d.clear_and_resize(1);
+    data->results_d.clear_and_resize(mem.results_memory / sizeof(char));
+    data->result_starts_d.clear_and_resize(n_alignments_initial + 1);
+    data->result_lengths_d.clear_and_resize(n_alignments_initial);
+    data->pvs            = batched_device_matrices<WordType>(mem.pmvs_matrix_memory / sizeof(WordType), allocator, stream);
```

Comment:
For whatever reason, when I call `reset_bandwidth` manually, it hangs here. When used via the ctor, it works fine.